packages/rest/src/router/routing-table.ts

```diff
@@ -281,6 +281,10 @@ export class ControllerRoute extends BaseRoute {
 
   updateBindings(requestContext: Context) {
     const ctor = this._controllerCtor;
+    requestContext
+      .bind('controller.current')
+      .toClass(ctor)
+      .inScope(BindingScope.SINGLETON);
```

Comment:
This does not right correct to me. IIUC, `SINGLETON` caches a single instance at the highest (app-wide) context. As a result, we will have only one controller instance shared by all incoming requests, which is goes against the current design where controllers can use properties to store temporary request-specific state.

Please revert this change.

Reply:
You probably misunderstand the usage here. The code binds the current controller as a singleton at request context level so that all references to the current controller for a given request will have a singleton.

Reply:
Interesting. So you are saying that `BindingScope.SINGLETON` does not apply to the top-most parent context in the chain, but only to the context where the binding was created. Could you please capture that in the documentation for SINGLETON scope, perhaps as an example?

Reply:
Also please add a comment here too, explaining that the SINGLETON scope applies to `requestContext` only.

Please make sure there is a test in `@loopback/context` covering this scenario (i.e. a parent "app" context, multiple child "request" contexts, a class bound as a singleton to each "request" context, and then verify that the singleton value is cached at the right level in the context chain).

Reply:
See https://github.com/strongloop/loopback-next/blob/master/packages/context/src/binding.ts#L64-L85. The tsdoc already describes it with an example.